Merge Conflict/Assets/Scripts/ComponentMovement.cs

```diff
@@ -0,0 +1,279 @@
+/**********************************************************************************************************************
+Name:          ComponentMovement
+Description:   Manages the Movement of the components if they are on the desk
+Author(s):     Simeon Baumann
+Date:          2024-03-14
+Version:       V1.1
+**********************************************************************************************************************/
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class ComponentMovement : MonoBehaviour
```

Comment:
Grundsätzlich okay, allerdings ist es schwierig durchzusehen und denke auch zu kompliziert als eigentlich notwendig. Da es aber funktioniert und wir in Zukunft wahrscheinlich daran auch nicht viel ändern werden, wäre es okay für mich wenn es grundlegend so bleibt. 
Die Entscheidung liegt bei dir. Ich schreib hir noch meine Änderungsvorschläge hin:

- when deciding to move, generate a destinationPosition
- when generating a destinationPosition check if in desk, when not, just retry generating
- in Update() just smoothly interpolate position to destinationPosition, no further logic 
- after dragging an element, ComponentHandler can give position before dragging, no need to keep array of last positions.
- apply same concept to scale (setting a scaleFactor and then just smoothly interpolating to  scaleFactor)
- generally have ComponentHandler.cs call functions in ComponentMovement.cs vs. setting bools

Reply:
smooth position follow reference (without offset): https://gamedev.stackexchange.com/a/120426

Reply:
In `CalculateNewDestination` wird eine die destinationPosition (`vectorToPositionInDirection`) berechnet und geschaut, dass sich mind. die Hälfte des Weges auf dem desk befindet. Dadurch können die Components bis zum Rand gelangen und der Platz kann voll und ganz ausgenutzt werden.

Die Abspeicherung der StartPosition beim draggen hatte ich auch überlegt. Nur kann es dann vorkommen, dass sich das Element in eine andere Richtung weiterbewegt, sobald es losgelassen wurde. Das fühlt sich dann nicht so natürlich an, weshalb ich hier in einem Array die letzten Positionen abspeichere.

Das Movement ist jetzt smooth. Ich hab die Methode aus dem Link übernommen.

Der ComponentHandler handlet jetzt das Movement der Componenten und ruft Methoden aus ComponentMovement.cs auf.



Bitte nochmal die beiden Dateien durchschauen, da sich die Aufrufe der Methoden verschoben haben.